### ISSUE TITLE
Insert of null-value via SQL API causes exclusion via ODocument API

### ISSUE THREAD

**Issue by deusaquilus** [orientechnologies/orientdb#9819]
## Expected behavior  
Inserting a null-value column like this:
```
DROP Class Contact
CREATE Class Contact
CREATE Property Contact.firstName STRING
CREATE Property Contact.lastName STRING (MANDATORY TRUE)
CREATE Property Contact.age INTEGER
CREATE Property Contact.addressFk INTEGER
INSERT INTO Contact SET firstName = 'Joe', lastName = null, age = 123, addressFk = 456
```
The querying the data like this:
```
oDatabase.query<java.util.List<ODocument>>("SELECT firstName, lastName, age, addressFk FROM Contact")
```
Should return an ODocument where:
```
// the document.fieldNames() should be:
List("firstName", "lastName", "age", "addressFk")
// the document.fieldValues() should be:
List("Joe", null, 123, 456)
```

## Actual behavior  
The returned document is missing the `lastName` value:
```
// the document.fieldNames() is:
List("firstName", "age", "addressFk")
// the document.fieldValues() is:
List("Joe", 123, 456)
```

## Steps to reproduce  
The above script:
```
DROP Class Contact
CREATE Class Contact
CREATE Property Contact.firstName STRING
CREATE Property Contact.lastName STRING (MANDATORY TRUE)
CREATE Property Contact.age INTEGER
CREATE Property Contact.addressFk INTEGER
INSERT INTO Contact SET firstName = 'Joe', lastName = null, age = 123, addressFk = 456
```
... and then the query `SELECT firstName, lastName, age, addressFk  FROM Contact` should be enough to reproduce the issue so long as it is loaded into an ODocument.

## Background

I maintain a Scala-based open source library called Quill (https://github.com/zio/zio-quill) that implements a language-integrate-query (i.e. LINQ) layer on top of several databases including OrientDB. My implementation relies on two things:

1. Being able to access columns in a result-set based on index as opposed to column name (this is why I cannot easily move to the `OResultSet`-based API).
2. Having RDBMS-like treatment of null-values. I.e. null-valued columns should be in the result-set.

My call-site for oDatabase.query is [here](https://github.com/zio/zio-quill/blob/42ceeff0f81659070688157950ec827c06c9c6a6/quill-orientdb/src/main/scala/io/getquill/OrientDBSyncContext.scala#L42). I then have a Decoding-stack [here](https://github.com/zio/zio-quill/blob/42ceeff0f81659070688157950ec827c06c9c6a6/quill-orientdb/src/main/scala/io/getquill/context/orientdb/encoding/Decoders.scala#L20) which decodes the value of a column by index. As a last-resort workaround null-valued issues I check to see whether the total number of columns I'm getting back from the query is the same number as I expect but this has many, many problems (e.g. what if the null-valued column is in the middle of the result-set and columns after it are in a datatype that I do not expect).

Bottom line, is there any way to get *all* the columns that I am querying for in the ODocument result?
If not, is there any OrientDB Java API that I can use to do that which support column-index based lookup?

Thanks!

P.S. This issue is pertinent to a recent large internal overhaul that I was recently doing:
https://github.com/zio/zio-quill/pull/2504#issuecomment-1144411504.
